exe/kubernetes-deploy

```diff
@@ -29,6 +30,10 @@ ARGV.options do |opts|
     "Timeout error is raised if it takes longer than the specified number of seconds") do |t|
     max_watch_seconds = t.to_i
   end
+  opts.on("--selector=SELECTOR", "Ensure that all resources in your template dir match the given selector, " \
+    "and restrict pruning to deployed resources it selects.  (format: k1=v1,k2=v2 or JSON)") do |s|
```

Comment:
One idea could be to make a really simple `LabelSelector` class that has `to_h` and `to_s` method and can be instantiated via `LabelSelector.from_string(the_arg_we_get_here)`, which would raise on malformed input like the bindings parser does. That would give us a place for the input validation and would still avoid duplicating the hash/string transformation logic, but without needing a utility module (those are a bit of a smell for me because in my experience they turn into a dumping ground of miscellany).